src/edit/control/Edit.Control.Delete.js

```diff
@@ -1,24 +1,32 @@
-L.EditToolbar.Delete = L.Handler.extend({
+L.Edit = L.Edit || {};
+L.Edit.Control = L.Edit.Control || {};
+
+L.Edit.Control.Delete = L.ToolbarAction.extend({
 	statics: {
 		TYPE: 'remove' // not delete as delete is reserved in js
 	},
 
+	options: {
+		toolbarIcon: { className: 'leaflet-draw-edit-remove' }
+	},
+
 	includes: L.Mixin.Events,
 
 	initialize: function (map, options) {
-		L.Handler.prototype.initialize.call(this, map);
+		L.ToolbarAction.prototype.initialize.call(this, map);
 
 		L.Util.setOptions(this, options);
 
 		// Store the selectable layer group for ease of access
 		this._deletableLayers = this.options.featureGroup;
+		this._map = map;
 
 		if (!(this._deletableLayers instanceof L.FeatureGroup)) {
 			throw new Error('options.featureGroup must be a L.FeatureGroup');
 		}
 
 		// Save the type so super can fire, need to do this as cannot do this.TYPE :(
-		this.type = L.EditToolbar.Delete.TYPE;
+		this.type = this.constructor.TYPE;
```

Comment:
This is pretty cool, I didn't know about [Object.prototype.constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/constructor)